spec/integration/type/file_spec.rb

```diff
@@ -695,22 +697,43 @@ def build_path(dir)
               :checksum => checksum_type,
               :source => env_path
             }
-            target_file = File.join(path, 'x')
+          end
 
-            first_catalog = Puppet::Resource::Catalog.new
-            first_catalog.add_resource Puppet::Type.send(:newfile, options)
-            first_catalog.apply
+          it "should not update the target directory" do
+            catalog.add_resource Puppet::Type.send(:newfile, @options)
+            catalog.apply
             expect(File).to be_directory(path)
             expect(Puppet::FileSystem.exist?(target_file)).to be_truthy
 
             # The 2nd time the resource should not change.
-            second_catalog = Puppet::Resource::Catalog.new
-            second_catalog.add_resource Puppet::Type.send(:newfile, options)
+            second_catalog.add_resource Puppet::Type.send(:newfile, @options)
             result = second_catalog.apply
             status = result.report.resource_statuses["File[#{target_file}]"]
             expect(status).not_to be_failed
             expect(status).not_to be_changed
           end
+
+          it "should update the target directory if contents change" do
+            pending "a way to appropriately mock ctime checks for a particular file" if checksum_type == 'ctime'
+
+            catalog.add_resource Puppet::Type.send(:newfile, @options)
+            catalog.apply
+            expect(File).to be_directory(path)
+            expect(Puppet::FileSystem.exist?(target_file)).to be_truthy
+
+            # Change the source file.
+            File.open(checksum_file, "wb") { |f| f.write "some content" }
+            File::Stat.any_instance.unstub(:mtime)
+            File::Stat.any_instance.unstub(:ctime)
```

Comment:
It would be good to eliminate the stubbing in the `with_checksum_types` shared context as that is only used in integration tests (so we should be able to create the necessary files). That might also fix the ctime issue (since we can delete the source file and recreate it, generating a new `ctime`)